manifests/config.pp

```diff
@@ -119,7 +123,7 @@
       serveraliases           => $foreman::serveraliases,
       server_port             => $foreman::server_port,
       server_ssl_port         => $foreman::server_ssl_port,
-      proxy_backend           => "http://${listen_socket}/",
+      proxy_backend           => "${foreman_service_bind}|http://127.0.0.1/",
```

Comment:
This is missing a port?

Reply:
As I understand it, this is required but is only used to determine the Host value sent with the request:

> In 2.4.7 and later, support for using a Unix Domain Socket is available by using a target which prepends unix:/path/lis.sock|. For example, to proxy HTTP and target the UDS at /home/www.socket, you would use unix:/home/www.socket|http://localhost/whatever/. Since the socket is local, the hostname used (in this case localhost) is moot, but it is passed as the Host: header value of the request.

Reply:
Ah, okay!

Reply:
But should it be localhost then, or the host the request came in on?

Reply:
> Since the current implementation depends on a Foreman change, we'll need to be careful about compatibility. Right now the module is compatible with Foreman 2.0, as specified in README.md. We want to support current stable + development.
> 
> Ideally we'd write any Foreman patch in such a way that it's compatible with both the old `FOREMAN_BIND` (without a protocol + port) and the new (with). Then cherry pick that into a stable branch. In the compatibility notes we'd then write that you need at least Foreman 2.1.x (2.1.2 at the time of writing?). I'd also like to cherry pick this specific patch to 2.1 when it's finished.

I'm not clear what you are asking for here. This change and the [foreman change](https://github.com/theforeman/foreman/pull/7972) go hand-in-hand and both would need to get back ported at the same time.